ipe_process_orchestrator/orchestrator.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import sys
+from typing import Any, Dict, NoReturn, Optional, Union
+import pandas as pd
+from ipe_utils.df_utils import df_columns_strip, df_remove_non_course_id
+from ipe_process_orchestrator.assignment_flow import IPEAssignmentFlow
+from api_handler.api_calls import APIHandler
+from constants import (COL_COURSE_ID)
+
+
+logger = logging.getLogger(__name__)
+
+
+class IPECompetenciesOrchestrator:
+
+    def __init__(self, props, original_df, api_handler) -> None:
+        """
+        Initialize the orchestrator
+        """
+        self.orginal_df: pd.DataFrame = original_df
+        self.props = props
+        self.api_handler: APIHandler = api_handler
+        self.filter_df_course_ids = pd.DataFrame()
+
+    def _clean_up_ipe_dataframe(self) -> None:
+        """
+        Clean up the dataframe
+        1. leading and trailing spaces df.columns
+        2. only courses id list with values that are numbers. Removes Shell, shell, empty, n/a, shell(23333)
+        3. the original dataframe will remain same and the filtered dataframe will be created with courseIds
+
+        """
+        try:
+            self.orginal_df.columns = df_columns_strip(self.orginal_df.columns)
+            cleaned_up_df = df_remove_non_course_id(self.orginal_df)
+            self.filter_df_course_ids = cleaned_up_df
+        except Exception as e:
+            logger.error(f'Error in clean_up_ipe_dataframe: {e}')
+            sys.exit(1)
+
+    def _create_delete_assignment(self, course: pd.Series) -> Union[NoReturn, str]:
+        """
+        Create the new IPE assignment if such assignment does not exist or delete the existing one if it exists. This action
+        is as a result of copied course and so delete the existing assignment is the best option.
+        """
+        course_id = course[COL_COURSE_ID]
+        rubric_id = self.props['rubric_id']
+        assignment_flow = IPEAssignmentFlow(
+            self.api_handler, course_id, rubric_id)
+        try:
+            assignment_id: str = assignment_flow.start_assignment_flow()
+            return assignment_id
+        except Exception as e:
+            raise e
+
+    def start_competencies_assigning_process(self, course: pd.Series) -> None:
```

Comment:
Looks like this function can be combined with the above _create_delete_assignment(), since it only makes call to that function?

Reply:
@zqian The next step in this process would be assigning competencies step as done [here](https://github.com/pushyamig/IPE-competencies-updater/blob/i14_assigning_competencies/ipe_process_orchestrator/orchestrator.py#L79), I will add that in later PR

Reply:
I have now added them as comments